File: src/main/java/com/fengsheng/handler/heart_tos.java
package com.fengsheng.handler;

import com.fengsheng.Game;
import com.fengsheng.HumanPlayer;
import com.fengsheng.network.ProtoServerChannelHandler;
import com.fengsheng.protos.Fengsheng;
import com.google.protobuf.GeneratedMessageV3;

public class heart_tos implements ProtoHandler {
    private static final short msgId = ProtoServerChannelHandler.stringHash("heart_toc");

    @Override
    public void handle(HumanPlayer player, GeneratedMessageV3 message) {
        byte[] buf = Fengsheng.heart_toc.newBuilder().setOnlineCount(Game.deviceCache.size()).build().toByteArray();
        player.send(msgId, buf, true);
    }
}

File: src/main/java/handler/heart_tos.kt
package com.fengsheng.handler

import com.fengsheng.Game
import com.fengsheng.HumanPlayer
import com.fengsheng.protos.Fengsheng

com.google.protobuf.GeneratedMessageV3
import java.util.concurrent.LinkedBlockingQueue
import io.netty.util.HashedWheelTimer

class heart_tos : ProtoHandler {
    override fun handle(player: HumanPlayer, message: GeneratedMessageV3) {
        val buf = Fengsheng.heart_toc.newBuilder().setOnlineCount(Game.Companion.deviceCache.size).build().toByteArray()
        player.send("heart_toc", buf, true)
    }
}